libs/queue/SPECIFICATION.md

```diff
@@ -4,32 +4,34 @@ This document provides an overview of the Queue library.
 
 It outlines the use cases, i.e. specification, and describes how to implement the library.
 
+The `Queue<T>` is a linear structure with a generic type parameter `T`. The elements in the queue are stored as a `Vec<T>`. Queues are growable, operate in a First-In-First-Out (FIFO) manner, and are created empty.
```

Comment:
This is a paragraph that really consists of 3 bullet points strung together. I would consider adjusting the format and presenting some heading / title to indicate what this is.

The overview begins by introducing the reader to the document itself and then suddenly into facts about the library. That's a large shift which can be eased by an appropriate grouping (heading / category etc.)